src/vma/sock/sockinfo_tcp.h

```diff
@@ -85,8 +85,23 @@ enum tcp_conn_state_e {
 	TCP_CONN_RESETED
 };
 
-typedef std::map<tcp_pcb*, int>		ready_pcb_map_t;
-typedef std::map<flow_tuple, tcp_pcb*>	syn_received_map_t;
+struct socket_option_t {
+	const int level;
+	const int optname;
+	const socklen_t optlen;
+	void *optval;
+
+	socket_option_t(const int _level, const int _optname, const void *_optval, const socklen_t _optlen) :
+		level(_level), optname(_optname), optlen(_optlen), optval(malloc(optlen)) {
+		memcpy(optval, _optval, optlen);
```

Comment:
do we need to use optval check before usage?

Reply:
I am checking it here : https://github.com/Mellanox/libvma/pull/704/files#diff-4ca769a22650c4fad33dfaaa485554e0R3692